web/core/components/project-states/state-item.tsx

```diff
@@ -0,0 +1,186 @@
+        await moveStatePosition(workspaceSlug, projectId, payload.id, payload);
+      } catch (error) {
+        console.error("error", error);
+      }
+    },
+    [workspaceSlug, projectId, moveStatePosition]
+  );
+
+  // derived values
+  const isDraggable = totalStates === 1 ? false : true;
+
+  // DND starts
+  // ref
+  const draggableElementRef = useRef<HTMLDivElement | null>(null);
+  // states
+  const [isDragging, setIsDragging] = useState(false);
+  const [isDraggedOver, setIsDraggedOver] = useState(false);
+  const [closestEdge, setClosestEdge] = useState<string | null>(null);
+  useEffect(() => {
+    const elementRef = draggableElementRef.current;
+    const initialData = { groupKey: groupKey, id: state.id, isFirstElement, isLastElement };
+
+    if (elementRef && state) {
+      combine(
+        draggable({
+          element: elementRef,
+          getInitialData: () => initialData,
+          onDragStart: () => setIsDragging(true),
+          onDrop: () => setIsDragging(false),
+          canDrag: () => isDraggable,
+        }),
+        dropTargetForElements({
+          element: elementRef,
+          getData: ({ input, element }) =>
+            attachClosestEdge(initialData, {
+              input,
+              element,
+              allowedEdges: ["top", "bottom"],
+            }),
+          onDragEnter: (args) => {
+            setIsDraggedOver(true);
+            setClosestEdge(extractClosestEdge(args.self.data));
+          },
+          onDragLeave: () => {
+            setIsDraggedOver(false);
+            setClosestEdge(null);
+          },
+          onDrop: (data) => {
+            setIsDraggedOver(false);
+            const { self, source } = data;
+            const sourceData = source.data;
+            const destinationData = self.data;
+
+            if (sourceData && destinationData && sourceData.id) {
+              const destinationGroupKey = destinationData.groupKey as TStateGroups;
+              const edge = extractClosestEdge(destinationData) || undefined;
+              const payload: Partial<IState> = {
+                id: sourceData.id as string,
+                group: destinationGroupKey,
+                sequence: getCurrentStateSequence(groupedStates[destinationGroupKey], destinationData, edge),
+              };
+              handleStateSequence(payload);
+            }
+          },
+        })
+      );
+    }
+  }, [
+    draggableElementRef,
+    state,
+    groupKey,
+    isDraggable,
+    groupedStates,
+    handleStateSequence,
+    isFirstElement,
+    isLastElement,
+  ]);
+  // DND ends
+
+  return (
+    <>
+      {updateStateModal ? (
+        <StateUpdate
+          workspaceSlug={workspaceSlug}
+          projectId={projectId}
+          state={state}
+          handleClose={() => setUpdateStateModal(false)}
+        />
+      ) : (
+        <>
+          <DropIndicator isVisible={isDraggedOver && closestEdge === "top"} />
+          <div
```

Comment:
It is looking hard to read, please add comments on the sections so we know what they are by reading them